packages/near-fast-auth-signer/src/components/CreateAccount/CreateAccount.tsx

```diff
@@ -160,6 +155,7 @@ function CreateAccount() {
     const methodNames = searchParams.get('methodNames');
 
     try {
+      if (!isValid) return;
```

Comment:
If we always call the createAccount with the handlerSubmit form hooks forms we are sure that it's a valid submission, so I believe we can remove this line.

Reply:
`createAccount` is also triggered by `useEffect`, which require condition to pause.

Reply:
In that case I think it's better to always call createAccount using handleSubmit(), so we can take advantage of the react-hooks-forms validation 

The way you implemented I believe the error won't be displayed to the user and we are duplicating code.

Reply:
I'm not sure if that's what we want. (I wasn't the one who implemented this part so I'm only guessing from previous implementation) 

I think having `useEffect` to run `createAccount` is actually good UX where, from previous screen, if email and account id is validated, user doesn't need to made fix and redirected to email verify page which is less of click. 

If we always force `handleSubmit` to do validation and click to go to next page, its another click of a button with extra screen. 

Personally, unless it is technically bad choice, I think current approach we have is good to move forward. (I always vote for better UX first)

Reply:
Oh no, we don't need to request the user to click the button.

It's just wrap the createAccount inside handleSubmit() inside the useEffect 

useEffect(() => {handleSub mit(createAccount)()}, [])

That way we make sure we run the validation before submitting. The email and username can come from the URL so the caller may not have run the validation before calling the fast-auth-signer

Reply:
agree! I have updated the changes and also tested that it work as expected

Reply:
Perfect, thanks Danny 😄